src/control/XournalMain.cpp

```diff
@@ -70,8 +71,23 @@ void XournalMain::initLocalisation() {
     std::cout.imbue(std::locale());
 }
 
+void XournalMain::migrateSettings() {
+    Path newConfigPath(g_get_user_config_dir());
+    newConfigPath /= g_get_prgname();
```

Comment:
`g_get_prgname()` is typically null here since the migration code is run before calling `gtk_init`. See
https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-get-prgname